src/task_queue/nudges/user_event_nudge.py

```diff
@@ -90,42 +94,78 @@ def get_email_lists(users):
             emails.append(user.email)
 
     return emails
-          
+
 
 def update_last_notification_dates(email):
     new_date = str(datetime.date.today())
-    user =  UserProfile.objects.filter(email=email).first()
+    user = UserProfile.objects.filter(email=email).first()
     user_notification_dates = user.notification_dates if user.notification_dates else {}
 
-    notification_dates = {**user_notification_dates,"user_event_nudge":new_date}
-    UserProfile.objects.filter(email=email).update(**{"notification_dates":notification_dates })
+    notification_dates = {**user_notification_dates, "user_event_nudge": new_date}
+    UserProfile.objects.filter(email=email).update(**{"notification_dates": notification_dates})
+
 
+def is_event_eligible(event, community_id, task=None):
+    now = timezone.now()
+    settings = event.nudge_settings.filter(communities__id=community_id).first()
+
+    if settings.never:
+        return False
+
+    freq = task.frequency if task else None
+    last_last_run = None
+
+    freq_to_delta = {
+        "EVERY_WEEK": relativedelta(weeks=1),
+        "bi-weekly": relativedelta(weeks=2),
+        "EVERY_MONTH": relativedelta(months=1),
+        "EVERY_DAY": relativedelta(days=1)
+    }
+
+    if freq:
+        last_last_run = now - freq_to_delta.get(freq, relativedelta(days=0))
```

Comment:
i'm not sure whether this might be a problem.  The event nudge task generally runs daily, though users are generally nudged once a week based on their preferences.

Reply:
Here is what I think the problem is:

A user last received a notification three days ago, with once a week nudge preferences, and the nudger runs every day at noon.  The event was published last night.   
Today when the nudger runs at noon, last_last_run is yesterday at noon, and if settings.when_first_posted is true, the function is_event_eligible will return true.  But the user won't be nudged since it hasn't been a week since the last nudge.
Then tomorrow when the nudger runs again, the last_last_run will be after the event was published, so it won't be sent out.

Reply:
It's true that we are thinking about creating separate celery jobs for event-specific notifications. This means that when an admin adds a setting for an event, we can create a celery job that will send reminders to users about that event only, in addition to the general event nudge that goes out to users every week.